ivadomed/training.py

```diff
@@ -54,6 +55,25 @@ def train(model_params, dataset_train, dataset_val, training_params, path_output
     # Write the metrics, images, etc to TensorBoard format
     writer = SummaryWriter(log_dir=path_output)
 
+    # Collect all hyperparameters into a dictionary
+    cfg = {
```

Comment:
I noticed some possible parameters are not included in the `cfg`, is that expected? (ex: `dropout_rate`, `final_activation`, `length_2D`)
I feel like adding params individually may be very high maintenance in the long term and prone to error (if the goal is to have everything).

Would it be possible to merge both the `training_params` dict and `model_params` dict into this `cfg`?

Reply:
> I feel like adding params individually may be very high maintenance in the long term and prone to error (if the goal is to have everything).

Thank you! Yes, the goal is to log as many hyperparameters as possible but while I was adding them 1-by-1 I felt that it is indeed error-prone (like the parameters you pointed out which I missed)! I'll look into the linked config manager file but it seems that merging both` training_params` and `model_params` is an efficient option. Thanks again for the pointers! :)

Reply:
> Would it be possible to merge both the training_params dict and model_params dict into this cfg?

that would be brilliant 🙌

Reply:
@naga-karthik so I wasn't sure if WandB have this, but Comet.ML actually picked up some parameters/metrics automatically when I was playing with its integration with tensorflow. In your experience, does WandB do some auto tracking once invoked in the context of Torch? The easiest way to usually tell is to log into WandB and see what is tracked vs what you told it to explicitly track. 

Also, it will probably be good to directly log entire fully overwritten JSON file once the myriad of configuration integration is complete.

Reply:
Thank you for the suggestion @mariehbourget! I could merge both `training_params` and `model_params` and the result now is that everything is being logged. It can be seen [here](https://wandb.ai/naga-karthik/ivado-wandb-testing/groups/temp_v2/table?workspace=user-naga-karthik) in a table form. Right now, I feel that the column names are too big (since they have been directly taken from the above dicts, I am looking into how to make it more informative.

@dyt811 So as far as the parameters/metrics related to your model are concerned, wandb logs them only when you tell it to. There are other external aspects as the environment, system, num_gpus, usage, etc. which it logs automatically (see "System" drop-down in [this](https://wandb.ai/naga-karthik/ivado-wandb-testing/groups/temp_v2/workspace?workspace=user-naga-karthik) link). In addition, it also logs some metadata and media which can be seen in "Files" in [this](https://wandb.ai/naga-karthik/ivado-wandb-testing/groups/temp_v2/files?workspace=user-naga-karthik) link.   

Also, could you please elaborate a bit more on this:

> directly log the entire fully overwritten config JSON

Right now, the input config file is already saved in the output folder after training, right? It will still be saved with the wandb keys added to the config file. Or, did you mean something else?